internal/featuregates/features.go

```diff
@@ -0,0 +1,59 @@
+package featuregates
```

Comment:
@beraldoleal sorry I didn't understand the message "changing the approach a "little bit"...". Can you please clarify ?

Reply:
Sorry @bpradipt, I just took a second look and it seems you are just setting the config per feature, so please ignore my comment here.

Reply:
With some assumptions let me try to answer it..
1. The reason for using package is for better code organisation. Since earlier what we had were just dummy features there was not enough thinking about how the feature gates will be used, how state management will happen etc. Now that real feature is being implemented, I took the liberty to organise the code in a modular fashion. I can rework to split this commit into two, but not sure if it's really needed. But if you insist I'll do it. 
2. There is a different logic to figure out the status of featuregates in the reconcile loop. The earlier logic only handled single feature and reading configmap  repeatedly for each individual feature is sub-optimal in the reconcile loop. Hence I use a map for the same. Let me know if this helps. I'll add this in the commit message as well

Reply:
Hi @bpradipt, improving the existing code is definitely the way to go here.

I recommend splitting the changes into separate commits. Merging them into another feature might cause some reviewers to miss them. I won't "insist", but it would be really helpful, if you don't mind.